src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp

```diff
@@ -2165,65 +2168,115 @@ namespace OpenMS
     updateMenu();
   }
 
-  void TOPPViewBase::showCurrentPeaksAsIonMobility()
-  {
-    double IM_BINNING = 1e5;
+  /**
+    @brief Split a (TimsTOF) ion mobility frame (i.e. a spectrum concatenated from multiple spectra with different IM values) into separate spectra
+   
+    The input @p im_frame must have a floatDataArray where IM values are annotated. If not, an exception is thrown.
 
-    const LayerData& layer = getActiveCanvas()->getCurrentLayer();
+    To get some coarser binning, choose a smaller @p number_of_bins. The default creates a new bin (=spectrum in the output) for each distinct ion mobility value.
+      
+    @param im_frame Concatenated spectrum representing a frame
+    @param number_of_bins In how many bins should ion mobility values be sliced? Default(-1) assigns all peaks with identical ion-mobility values to a separate spectrum.
+    @return IM frame split into multiple bins (= 1 spectrum per bin)
 
-    // Get current spectrum
-    auto spidx = layer.getCurrentSpectrumIndex();
-    MSSpectrum tmps = layer.getCurrentSpectrum();
+    @throws Exception::MissingInformation if @p im_frame does not have IM data in floatDataArrays
+  */
+  MSExperiment splitByIonMobility(MSSpectrum im_frame, UInt number_of_bins = -1)
```

Comment:
sounds a bit like what we do for FAIMS. could this be moved out of TOPPView?

Reply:
yes!
but currently the `hack` which overwrites RT with IM is needed.
Will move when its fixed.